src/app/services/products.service.ts

```diff
@@ -0,0 +1,27 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { IProduct } from '@shared/types';
+import { environment as env } from './../../environments/environment.development';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProductsService {
+
+  constructor(
+    private http: HttpClient,
+  ) { }
+
+  getAllProducts(): Observable<IProduct[]> {
+    return this.http.get<IProduct[]>(env.BASE_JSON_URL);
+  }
+
+  fetchData(param: string): Observable<IProduct[]> {
```

Comment:
It's better to use the built-in HttpParams() functionality:
const params = new HttpParams().set('product_type', `${param}`).set('otherKey', 'otherValue')
return this.http.get<IProduct[]>(`${env.BASE_JSON_URL}`, { params })


There is no need to store parameters in the environment, except that the parameter will be different depending on the environment.